backend/service/audit/storage/sql/sql.go

```diff
@@ -187,6 +187,28 @@ func (c *client) query(ctx context.Context, query string, args ...interface{}) (
 	return events, nil
 }
 
+func (c *client) AttemptLock(ctx context.Context, lockID uint32) (bool, error) {
+	var lock bool
+	if err := c.db.QueryRowContext(ctx, "SELECT pg_try_advisory_lock($1);", lockID).Scan(&lock); err != nil {
+		c.logger.Error("Unable to query for a advisory lock", zap.Error(err))
+		return false, err
+	}
+	return lock, nil
+}
+
+func (c *client) ReleaseLock(ctx context.Context, lockID uint32) (bool, error) {
+	var unlock bool
+	if err := c.db.QueryRowContext(ctx, "SELECT pg_advisory_unlock($1)", lockID).Scan(&unlock); err != nil {
+		c.logger.Error("Unable to perform an advisory unlock", zap.Error(err))
+		return false, err
+	}
+	return unlock, nil
+}
+
+func (c *client) GetLockID(lockID string) uint32 {
+	return storage.ConvertLockToUint32(lockID)
+}
```

Comment:
this function being on client doesn't really seem to add anything, just call the Convert func directly?

Reply:
My initial though was that other providers might want to compute locks differently, however the interface still requires you to return a `unit32` so yeah I think it's a moot point. Let me remove.